src/rgw/driver/rados/rgw_bl_rados.cc

```diff
@@ -0,0 +1,786 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "rgw_bucket_logging.h"
+#include "rgw_bl_rados.h"
+#include <chrono>
+#include <fmt/format.h>
+#include <future>
+#include <memory>
+#include <thread>
+#include <boost/algorithm/hex.hpp>
+#include <boost/asio/basic_waitable_timer.hpp>
+#include <boost/asio/executor_work_guard.hpp>
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/spawn.hpp>
+#include <boost/context/protected_fixedsize_stack.hpp>
+#include "common/async/yield_context.h"
+#include "common/async/yield_waiter.h"
+#include "common/ceph_time.h"
+#include "common/dout.h"
+#include "cls/lock/cls_lock_client.h"
+#include "include/common_fwd.h"
+#include "include/function2.hpp"
+#include "librados/AioCompletionImpl.h"
+#include "services/svc_zone.h"
+#include "rgw_common.h"
+#include "rgw_perf_counters.h"
+#include "rgw_sal_rados.h"
+#include "rgw_zone_features.h"
+
+
+#define dout_subsys ceph_subsys_rgw_bucket_logging
+
+namespace rgw::bucketlogging {
+
+using commit_list_t = std::set<std::string>;
+
+// use mmap/mprotect to allocate 128k coroutine stacks
+auto make_stack_allocator() {
+  return boost::context::protected_fixedsize_stack{128*1024};
+}
+
+const std::string COMMIT_LIST_OBJECT_NAME = "bucket_logging_global_commit_list";
+
+static std::string get_commit_list_name (rgw::sal::Bucket *log_bucket,
+                                         const std::string& prefix) {
+  return  log_bucket->get_tenant() + "/" + log_bucket->get_name() + "/"
```

Comment:
please avoid the plus operator on strings, as it creates quite a bit of temporary strings.
use `std::format` instead

Reply:
Done.